crates/account_sdk/src/session_token/session.rs

```diff
@@ -101,15 +101,11 @@ impl Session {
             .map(|c| self.call_proof(c).ok_or(SessionError::CallNotInPolicy))
             .collect::<Result<Vec<_>, SessionError>>()?;
 
-        let proofs_flat = proofs.into_iter().fold(Ok(vec![]), |acc, proof| {
-            acc.and_then(|mut acc| {
-                acc.extend(proof.1.clone());
-                Ok(acc)
-            })
+        let proofs_flat = proofs.into_iter().try_fold(vec![], |mut acc, proof| {
+            acc.extend(proof.1.clone());
+            Ok(acc)
         })?;
 
-        assert!(!self.session_token().is_empty(), "Session token is empty");
-
         Ok(SessionSignature {
```

Comment:
Which change induced this removal?

Reply:
Sorry, my bad, i've reverted this now. I was making cargo clippy happy and must've accidentaly removed the line.